mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -1356,8 +1356,10 @@ ParseResult MapOp::parse(OpAsmParser &parser, OperationState &result) {
     return failure();
 
   if (payloadOpName.has_value()) {
-    addBodyWithPayloadOp(parser, result, payloadOpName.value(), payloadOpAttrs,
-                         ArrayRef(result.operands).drop_back());
+    if (!result.operands.empty())
```

Comment:
nit: Can you try something like this instead of guarding it with an `if` check?
```c++
      addBodyWithPayloadOp(parser, result, payloadOpName.value(),
                           payloadOpAttrs,
                           result.operands.empty() ? ArrayRef(result.operands) : ArrayRef(result.operands).drop_back())
```

I don't know if it will work, but maybe we can get a better error message than:
```mlir
  // expected-error @+1 {{'linalg.map' op requires one region}}
  linalg.map { arith.addf }
```
(We did specify a region, we just didn't parse it.)

Reply:
Alternatively: I think adding an empty region in the `else` branch should give a better error message.

Reply:
> I don't know if it will work, but maybe we can get a better error message than:

Passing an empty `ArrayRef` unfortunately doesn't work, because during creation of the region, the Element Type of the respective tensor is needed.
But adding an empty region improves the error message.